R/finalize_plot.R

```diff
@@ -554,8 +572,10 @@ buildChart <- function(plot,
   if (number_of_legends > 1) {
     # if multilegend plot, establish loop to change margins
     for (i in 1:(number_of_legends-1)) {
-      margin_index <- 2*(i+1) # e.g., for a 2-legend item, this modifies element 4
-      legend$heights[[margin_index]] <- grid::unit(consts$margin_legend_i,"bigpts")
+      # e.g., for a 2-legend item, this modifies element 4
+      margin_index <- 2*(i+1)
+      # apply correct legend spacing
+      legend$heights[[margin_index]] <- grid::unit(margin_legend_i,"bigpts")
```

Comment:
Presuming this is already set correctly barring an override, this whole section could be shifted up into the if override exists section